src/main/java/com/ozonehis/eip/odooopenmrs/client/OdooClient.java

```diff
@@ -0,0 +1,133 @@
+package com.ozonehis.eip.odooopenmrs.client;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.singletonList;
+import static java.util.Collections.singletonMap;
+
+import com.ozonehis.eip.odooopenmrs.Constants;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.*;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.xmlrpc.XmlRpcException;
+import org.apache.xmlrpc.client.XmlRpcClient;
+import org.apache.xmlrpc.client.XmlRpcClientConfigImpl;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Getter
+@Component
+public class OdooClient {
+
+    @Value("${odoo.baseUrl}")
+    private String url;
+
+    @Value("${odoo.database}")
+    private String database;
+
+    @Value("${odoo.username}")
+    private String username;
+
+    @Value("${odoo.password}")
+    private String password;
+
+    private Integer uid;
+
+    private XmlRpcClient client;
+
+    private static final String SERVER_OBJECT_URL = "%s/xmlrpc/2/object";
+
+    private static final String SERVER_COMMON_URL = "%s/xmlrpc/2/common";
+
+    public OdooClient() {}
+
+    public void init() throws IOException {
+        client = new XmlRpcClient() {
+            {
+                setConfig(new XmlRpcClientConfigImpl() {
+
+                    {
+                        setEnabledForExtensions(true);
+                        setServerURL(new URL(String.format(SERVER_OBJECT_URL, getUrl())));
+                    }
+                });
+            }
+        };
+    }
+
+    public void authenticate() throws MalformedURLException {
+        // TODO: Where should we place the init logic
+        try {
+            init();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
I'd probably do this in the opposite way, i.e., instead of `authenticateIfNecessary()` each function should call `init()` which should:

1. Initialize the client if it hasn't already been initialized
2. Authenticate the client if it hasn't already been authenticated